cterasdk/edge/network.py

```diff
@@ -92,28 +93,42 @@ def enable_dhcp(self):
 
         logging.getLogger().info('Network settings updated. Enabled DHCP.')
 
-    def tcp_connect(self, address, port):
+    def diagnose(self, services):
+        """
+        Test a TCP connection to a host over a designated port
+
+        :param list[cterasdk.edge.types.TCPService] services: List of services, identified by a host and a port
+        :returns: A list of named-tuples including the host, port and a boolean value indicating whether TCP connection can be established
+        :rtype: list[cterasdk.edge.types.TCPConnectResult]
+        """
+        results = []
```

Comment:
This can be a one liner:
return [self.tcp_connect(service) for service in services]

Reply:
done